### PR TITLE
Code splitting based on routes

### PR DESCRIPTION
**Pull request by rs-8**
Hello!
Introduce code splitting based on routes

Before:
Screenshot 2024-05-18 at 20 22 56
Screenshot 2024-05-18 at 20 21 26

After:
Screenshot 2024-05-18 at 15 30 26
Screenshot 2024-05-18 at 20 24 14

Thanks!

### CODE REVIEW

File: src/pages/layout/index.tsx [hunk 1]
```diff
@@ -133,7 +133,7 @@ export default function ExplorerLayout({children}: LayoutProps) {
             >
               <Header />
               <Container maxWidth="xl" sx={{flexGrow: 4, paddingTop: "2rem"}}>
-                {children}
+                <Suspense>{children}</Suspense>
```
Comment (blakezimmerman) [inline src/pages/layout/index.tsx:136]: Can we add a `fallback` here so that something is shown while the page chunk loads?

Comment (rs-8) [reply src/pages/layout/index.tsx:136]: Yes, of course, but I saw that in the case of a fast network we get flickering if we use CircularProgress for all pages, another option is to make a individual skeleton for each page (difficult), so I left it without a fallback


https://github.com/aptos-labs/explorer/assets/26749651/a7a82768-18b2-4e5d-9250-7274db5ce76d

Comment (rs-8) [reply src/pages/layout/index.tsx:136]: I also see that most of the application is loaded gradually without loaders, and in this approach it looks better to me, I could be wrong

Comment (blakezimmerman) [reply src/pages/layout/index.tsx:136]: Yeah, you definitely raise a good point regarding the quick flash of a loading spinner. One approach you could take is to write a fallback component that only shows the spinner after a timeout has elapsed. The following is just an example and you may want to tweak the timeout based on what feels right.

```tsx
function Fallback() {
  const [fallback, setFallback] = useState<ReactNode | null>(null);

  useEffect(() => {
    const timeout = window.setTimeout(() => setFallback(<CircularProgress />), 150);
    return () => {
      window.clearTimeout(timeout);
    };
  }, []);
  
  return fallback;
}
```

Comment (rs-8) [reply src/pages/layout/index.tsx]: Added a fallback with a timeout, good solution, thanks for the advice

**Review by blakezimmerman — APPROVED** [review]

**Comment by rs-8** [comment]
@blakezimmerman 
Hello!
Can we merge pr?

**Comment by blakezimmerman** [comment]
> @blakezimmerman Hello! Can we merge pr?

Yeah, I think this is ready to go.
@kent-white Should we merge it?

**Comment by kent-white** [comment]
@rs-8 great job!

**Pull request merged**